plugin/src/main/groovy/com/novoda/staticanalysis/internal/detekt/DetektConfigurator.groovy

```diff
@@ -0,0 +1,80 @@
+package com.novoda.staticanalysis.internal.detekt
+
+import com.novoda.staticanalysis.StaticAnalysisExtension
+import com.novoda.staticanalysis.internal.Configurator
+import com.novoda.staticanalysis.internal.Violations
+import org.gradle.api.NamedDomainObjectContainer
+import org.gradle.api.Project
+import org.gradle.api.Task
+
+class DetektConfigurator implements Configurator {
+
+    private final Project project
+    private final Violations violations
+    private final Task evaluateViolations
+
+    static DetektConfigurator create(Project project,
+                                     NamedDomainObjectContainer<Violations> violationsContainer,
+                                     Task evaluateViolations) {
+        Violations violations = violationsContainer.maybeCreate('Detekt')
+        return new DetektConfigurator(project, violations, evaluateViolations)
+    }
+
+    private DetektConfigurator(Project project, Violations violations, Task evaluateViolations) {
+        this.project = project
+        this.violations = violations
+        this.evaluateViolations = evaluateViolations
+    }
+
+    @Override
+    void execute() {
+        project.extensions.findByType(StaticAnalysisExtension).ext."detekt" = { Closure config ->
+
+            if (!isKotlinProject(project)) {
+                return
+            }
+
+            project.apply plugin: 'io.gitlab.arturbosch.detekt'
+
+            project.extensions.findByName('detekt').with {
+                // apply configuration closure to detekt extension
+                config.delegate = it
+                config()
+            }
+
+            if (project.tasks.findByName('detektCheck')) {
+                configureToolTask()
+            }
+        }
+    }
+
+    private void configureToolTask() {
+        def detektTask = project.tasks.findByName('detektCheck')
+        // run detekt as part of check
+        project.tasks.findByName('check').dependsOn(detektTask)
+
+        // evaluate violations after detekt
+        detektTask.group = 'verification'
+        def collectViolations = createCollectViolationsTask(violations)
+        evaluateViolations.dependsOn collectViolations
+        collectViolations.dependsOn detektTask
+    }
+
+    private CollectDetektViolationsTask createCollectViolationsTask(Violations violations) {
+        def reportFilePath = "${project.extensions.findByName('detekt').systemOrDefaultProfile().output}/detekt-checkstyle.xml"
```

Comment:
`systemOrDefaultProfile` seems very specific. I looked at the source code and saw that they have `profiles` set. And this method is just searching for 1 specific profile in that set. 

What about looping over all possible profiles? Something like: 

```
def detekt = project.extensions['detekt']
detekt.profiles.each { profile ->
  def reportFile = new File(profile.output, "detekt-checkstyle.xml")
  if (reportFile.exists) {
    // create tasks
  }
}
```

Reply:
But i's `private`. Can we still access it?

Reply:
Doesn't look like `Could not get unknown property 'profiles' for DetektExtension` :(

Reply:
That's too bad. :( I wonder what happens when the user has different profiles set-up?
Anyways, this will cover the regular usage. Forget about my comment. 😄

Reply:
Maybe we should open an issue or a PR on Detekt for this? Looks like a reasonable tooling use case

Reply:
Good point, will add this to a ticket.

Reply:
I opened a ticket in Detekt https://github.com/arturbosch/detekt/issues/705